api/v1alpha1/eventing_types.go

```diff
@@ -48,47 +49,51 @@ const (
 
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// Eventing is the Schema for the eventing API.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="State",type="string",JSONPath=".status.state",description="State of Eventing"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="Age of the resource"
+type Eventing struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:default:={backends:{{type:""},{type:"NATS", config:{natsStreamStorageType:"File", natsStreamReplicas:3, natsStreamMaxSize:"700Mi", natsMaxMsgsPerTopic:1000000}}}, logging:{logLevel:Info}, publisher:{replicas:{min:2,max:2}, resources:{limits:{cpu:"500m",memory:"512Mi"}, requests:{cpu:"10m",memory:"256Mi"}}}}
+	Spec   EventingSpec   `json:"spec,omitempty"`
+	Status EventingStatus `json:"status,omitempty"`
+}
+
+// EventingStatus defines the observed state of Eventing
+type EventingStatus struct {
+	State      string             `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
 // EventingSpec defines the desired state of Eventing
 type EventingSpec struct {
 	// Backends defines the list of eventing backends to provision.
+	// +kubebuilder:default:={{type:""},{type:"NATS", config:{natsStreamStorageType:"File", natsStreamReplicas:3, natsStreamMaxSize:"700Mi", natsMaxMsgsPerTopic:1000000}}}
+	// +kubebuilder:validation:XValidation:rule=" (self[0].type != 'EventMesh') || ((self[0].type == 'EventMesh') && (self[0].config.eventMeshSecret != ''))", message="secret cannot be empty if EventMesh backend is used"
```

Comment:
I see it is only for one backend. I understand, maybe we should note it somewhere or create an issue not to forget that we should change the validation as we enable multiple backend support. What do you think?